hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobFileCleanerChore.java

```diff
@@ -256,9 +274,11 @@ public void cleanupObsoleteMobFiles(Configuration conf, TableName table) throws
         while (rit.hasNext()) {
           LocatedFileStatus lfs = rit.next();
           Path p = lfs.getPath();
-          if (!allActiveMobFileName.contains(p.getName())) {
-            // MOB is not in a list of active references, but it can be too
-            // fresh, skip it in this case
+          String[] mobParts = p.getName().split("_");
+          String regionName = mobParts[mobParts.length - 1];
+
+          if (!regionNames.contains(regionName)) {
```

Comment:
I know we talked about this before, but am a bit confused now on why we can't simply rely on the `allActiveMobFileName` list of all existing mob references built previously? 

And there's also the case of potentially archived split/merge parents containing files with mob refs. It seems both now and before this PR we were only tracking mob refs on non-archived regions, or am I missing something here?

Reply:
`allActiveMobFileName` does not contain files currently being written. MobFileCleanerChore is restricted by  `!regionNames.contains(regionName)` to only clean mob files where the creating region is already archived and we do not have to care about currently written files.

AFAIK after split/merge there is a minor compaction that gets rid of the hfile references and makes sure we have an actual hfile containing the mob file refs in the child regions before the parent would be archived. So we should always have a hfile referencing the mob files. But I'll verify this.

Reply:
>AFAIK after split/merge there is a minor compaction that gets rid of the hfile references and makes sure we have an actual hfile containing the mob file refs in the child regions before the parent would be archived. So we should always have a hfile referencing the mob files. But I'll verify this.

I know compaction is triggered, but ain't sure it will always run or complete (what if there were not many hfiles in the parent?), as it's not uncommon to see hfile references on store dirs. And how about snapshots clones, that could have references pointing to archive folder for a while?

Reply:
I think these are valid concerns and should be investigated, but maybe not as part of this issue. For now I'm following the original logic when looking for mob references and I hope that would be acceptable.

Reply:
Agreed. No need to address those here, better to have an specific ticket for those.